core/logger.go

```diff
@@ -0,0 +1,51 @@
+package core
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type LoggerConfig struct {
+	LogLevel            hclog.Level
+	JSONLogFormat       bool
+	OpenOrCreateNewFile bool
+	LogsDirectory       string
+	LogFile             string
+	Name                string
+}
+
+func NewLogger(config LoggerConfig) (l hclog.Logger, err error) {
+	var logFileWriter *os.File
+
+	if config.LogFile != "" {
+		fullFilePath := config.LogFile
```

Comment:
Can we merge these two fields
```
LogsDirectory       string
LogFile             string
```
into one `LogFilePath`?
I suggest extracting directory from path with `file := filepath.Base(path)` or similar